spopt/region/skater.py

```diff
@@ -0,0 +1,409 @@
+
+        if (labels is None) != (target_label is None):
+            raise ValueError(
+                "Both labels and target_label must be supplied! Only {} provided.".format(
+                    ["labels", "target_label"][int(target_label is None)]
+                )
+            )
+        if verbose:
+            try:
+                from tqdm import tqdm
+            except ImportError:
+
+                def tqdm(noop, desc=""):
+                    return noop
+
+        else:
+
+            def tqdm(noop, desc=""):
+                return noop
+
+        zero_in = (labels is not None) and (target_label is not None)
+        current_n_subtrees, current_labels = cg.connected_components(
+            MSF, directed=False
+        )
+        best_deletion = deletion(np.nan, np.nan, np.inf)
+        for in_node, out_node in tqdm(
+            np.vstack(MSF.nonzero()).T, desc="finding cut..."
+        ):  # iterate over MSF edges
+            if zero_in:
+                if labels[in_node] != target_label:
+                    continue
+
+            local_MSF = copy.deepcopy(MSF)
+
+            # delete a candidate edge
+            local_MSF[in_node, out_node] = 0
+            local_MSF.eliminate_zeros()
+
+            # get the connected components
+            local_n_subtrees, local_labels = cg.connected_components(
+                local_MSF, directed=False
+            )
+
+            if local_n_subtrees <= current_n_subtrees:
+                raise Exception("Malformed MSF!")
+
+            # compute the score of these components
+            score = self.score(data, labels=local_labels, quorum=quorum)
+
+            # if the score is lower than the best score and quorum is met
+            if score < best_deletion.score:
+                best_deletion = deletion(in_node, out_node, score)
+        if make:
+            return self.make_cut(*best_deletion, MSF=MSF)
+        return best_deletion
+
+    def make_cut(self, in_node, out_node, score, MSF=None):
+        """
+        make a cut on the MSF inplace, provided the in_node, out_node, MSF, and score. 
+        in_node: int, ID of the source node for the edge to be cut
+        out_node: int, ID of the destination node for the edge to be cut
+        score: float, the value of the score being cut. if the score is infinite, the cut is not made. 
+        MSF: the spanning forest to use when making the cut. If not provided,
+             uses the defualt tree in self.minimum_spanning_forest_
+        """
+        if MSF is None:
+            MSF = self.minimum_spanning_forest_
+        if np.isfinite(score):
+            MSF[in_node, out_node] = 0
+            MSF.eliminate_zeros()
+            return (MSF, *cg.connected_components(MSF, directed=False))
+        raise OptimizeWarning(
+            "Score of the ({},{}) cut is inf, the quorum is likely not met!"
+        )
+
+
+class Skater(BaseSpOptHeuristicSolver):
+    """...Needs a short description..."""
```

Comment:
Let's go ahead and fill in the description here and the docstrings in `__init__`.